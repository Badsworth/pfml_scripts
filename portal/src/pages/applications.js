import React, { useEffect } from "react";
import ApplicationCard from "../components/ApplicationCard";
import DashboardNavigation from "../components/DashboardNavigation";
import Heading from "../components/Heading";
import PropTypes from "prop-types";
import Spinner from "../components/Spinner";
import Title from "../components/Title";
import { useRouter } from "next/router";
import { useTranslation } from "../locales/i18n";

/**
 * List of all applications associated with the authenticated user
 */
const Applications = (props) => {
  const { appLogic } = props;
  const { t } = useTranslation();
  const router = useRouter();

  useEffect(() => {
    appLogic.loadClaims();
  });

  const claimsLoaded = !!appLogic.claims;
  const hasClaims = claimsLoaded && appLogic.claims.items.length > 0;
  const hasInProgressClaims =
    hasClaims && appLogic.claims.inProgress.length > 0;
  const hasSubmittedClaims = hasClaims && appLogic.claims.submitted.length > 0;

  return (
    <React.Fragment>
      <DashboardNavigation activeHref={router.route} />
      <Title hidden={hasClaims}>{t("pages.applications.title")}</Title>

      {claimsLoaded && !hasClaims && <p>{t("pages.applications.noClaims")}</p>}

      {!claimsLoaded && (
        <div className="margin-top-8 text-center">
          <Spinner aria-valuetext={t("components.spinner.label")} />
        </div>
      )}

      {hasInProgressClaims && (
        <React.Fragment>
          <Heading level="2" size="1">
            {t("pages.applications.inProgressHeading")}
          </Heading>
          {appLogic.claims.inProgress.map((claim, index) => (
            <ApplicationCard
              key={claim.application_id}
              claim={claim}
              number={index + 1}
            />
          ))}
        </React.Fragment>
      )}

      {hasSubmittedClaims && (
        <React.Fragment>
          <Heading level="2" size="1">
            {t("pages.applications.submittedHeading")}
          </Heading>
          {appLogic.claims.submitted.map((claim, index) => (
            <ApplicationCard
              key={claim.application_id}
              claim={claim}
              number={appLogic.claims.inProgress.length + index + 1}
            />
          ))}
        </React.Fragment>
      )}
    </React.Fragment>
  );
};

Applications.propTypes = {
  appLogic: PropTypes.object.isRequired,
};

export default Applications;
