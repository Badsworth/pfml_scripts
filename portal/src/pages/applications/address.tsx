import React, { useEffect } from "react";
import withBenefitsApplication, {
  WithBenefitsApplicationProps,
} from "../../hoc/withBenefitsApplication";
import AddressModel from "../../models/Address";
import ConditionalContent from "../../components/ConditionalContent";
import FieldsetAddress from "../../components/FieldsetAddress";
import InputChoiceGroup from "../../components/core/InputChoiceGroup";
import QuestionPage from "../../components/QuestionPage";
import { pick } from "lodash";
import useFormState from "../../hooks/useFormState";
import useFunctionalInputProps from "../../hooks/useFunctionalInputProps";
import { useTranslation } from "../../locales/i18n";

export const fields = [
  "claim.has_mailing_address",
  "claim.residential_address.line_1",
  "claim.residential_address.line_2",
  "claim.residential_address.city",
  "claim.residential_address.state",
  "claim.residential_address.zip",
  // Include `mailing_address` so validation error shows for completely empty mailing address.
  // We don't need this for `residential_address` since that defaults to a blank object, rather than null.
  "claim.mailing_address",
  "claim.mailing_address.line_1",
  "claim.mailing_address.line_2",
  "claim.mailing_address.city",
  "claim.mailing_address.sate",
  "claim.mailing_address.zip",
];

export const Address = (props: WithBenefitsApplicationProps) => {
  const { appLogic, claim } = props;
  const { t } = useTranslation();

  const { formState, getField, updateFields, clearField } = useFormState(
    pick(props, fields).claim
  );

  const { has_mailing_address } = formState;

  /**
   * When user indicates they have a mailing address,
   * add a blank mailing address so validations are ran against it
   */
  useEffect(() => {
    const existingMailingAddress = formState.mailing_address;
    if (formState.has_mailing_address && !existingMailingAddress) {
      updateFields({ mailing_address: {} });
    }
  }, [formState, updateFields]);

  const handleSave = () =>
    appLogic.benefitsApplications.update(claim.application_id, formState);

  const getFunctionalInputProps = useFunctionalInputProps({
    appErrors: appLogic.appErrors,
    formState,
    updateFields,
  });

  const residentialAddressProps = getFunctionalInputProps(
    "residential_address"
  );
  if (!residentialAddressProps.value) {
    residentialAddressProps.value = new AddressModel({});
  }

  const mailingAddressProps = getFunctionalInputProps("mailing_address");
  if (!mailingAddressProps.value) {
    mailingAddressProps.value = new AddressModel({});
  }

  return (
    <QuestionPage title={t("pages.claimsAddress.title")} onSave={handleSave}>
      <FieldsetAddress
        appErrors={appLogic.appErrors}
        label={t("pages.claimsAddress.sectionLabel")}
        hint={t("pages.claimsAddress.hint")}
        {...residentialAddressProps}
      />
      <InputChoiceGroup
        {...getFunctionalInputProps("has_mailing_address")}
        choices={[
          {
            checked: has_mailing_address === false,
            label: t("pages.claimsAddress.choiceYes"),
            value: "false",
          },
          {
            checked: has_mailing_address === true,
            label: t("pages.claimsAddress.choiceNo"),
            value: "true",
          },
        ]}
        label={t("pages.claimsAddress.hasMailingAddressLabel")}
        hint={t("pages.claimsAddress.hasMailingAddressHint")}
        type="radio"
      />
      <ConditionalContent
        fieldNamesClearedWhenHidden={["mailing_address"]}
        getField={getField}
        clearField={clearField}
        updateFields={updateFields}
        visible={has_mailing_address}
      >
        <FieldsetAddress
          appErrors={appLogic.appErrors}
          label={t("pages.claimsAddress.mailingAddressLabel")}
          hint={t("pages.claimsAddress.mailingAddressHint")}
          addressType="mailing"
          {...mailingAddressProps}
        />
      </ConditionalContent>
    </QuestionPage>
  );
};

export default withBenefitsApplication(Address);
