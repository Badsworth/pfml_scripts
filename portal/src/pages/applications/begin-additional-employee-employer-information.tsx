import withBenefitsApplication, {
  WithBenefitsApplicationProps,
} from "../../hoc/withBenefitsApplication";

import BackButton from "src/components/BackButton";
import ButtonLink from "src/components/ButtonLink";
import FormLabel from "src/components/core/FormLabel";
import Heading from "src/components/core/Heading";
import React from "react";
import Title from "src/components/core/Title";
import routes from "src/routes";
import { useTranslation } from "../../locales/i18n";

export const fields = [];

/**
 * A form page to begin collecting additional information for an employee/employer match.
 */
export const NoEeErMatchBegin = (props: WithBenefitsApplicationProps) => {
  const { appLogic, claim } = props;
  const { t } = useTranslation();

  return (
    <>
      <BackButton />
      <Title small={true}>{t("pages.claimsEmploymentStatus.title")}</Title>
      <Heading level="2" size="1">
        {t(
          "pages.noEmployeeEmployerMatchFlow.beginAdditionalEmployeeEmployerInformationTitle"
        )}
      </Heading>
      <p className="display-block line-height-sans-5 measure-5 usa-intro">
        {t(
          "pages.noEmployeeEmployerMatchFlow.beginAdditionalEmployeeEmployerInformationDescription"
        )}
      </p>
      <ButtonLink
        className="display-inline-flex flex-align-center flex-justify-center flex-column margin-right-0 margin-top-3"
        href={appLogic.portalFlow.getNextPageRoute(
          "CONTINUE",
          {},
          { claim_id: claim.application_id }
        )}
      >
        <div>
          {t(
            "pages.noEmployeeEmployerMatchFlow.beginAdditionalEmployeeEmployerInformationButton"
          )}
        </div>
      </ButtonLink>
    </>
  );
};

export default withBenefitsApplication(NoEeErMatchBegin);
