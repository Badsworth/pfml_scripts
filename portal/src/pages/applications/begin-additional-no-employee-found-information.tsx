import withBenefitsApplication, {
  WithBenefitsApplicationProps,
} from "../../hoc/withBenefitsApplication";

import BackButton from "src/components/BackButton";
import ButtonLink from "src/components/ButtonLink";
import Heading from "src/components/core/Heading";
import React from "react";
import Title from "src/components/core/Title";
import { useTranslation } from "../../locales/i18n";

export const fields = [];

/**
 * A form page to begin collecting additional information for an employee/employer match.
 */
export const AdditionalUserNotFoundInfoBegin = (
  props: WithBenefitsApplicationProps
) => {
  const { appLogic, claim } = props;
  const { t } = useTranslation();

  return (
    <React.Fragment>
      <BackButton />
      <Title small={true}>{t("pages.claimsEmploymentStatus.title")}</Title>
      <Heading level="2" size="1">
        {t(
          "pages.claimsAdditionalUserNotFoundInfo.beginAdditionalNoEmployeeFoundInfoTitle"
        )}
      </Heading>
      <p className="display-block line-height-sans-5 measure-5 usa-intro">
        {t(
          "pages.claimsAdditionalUserNotFoundInfo.beginAdditionalNoEmployeeFoundInfoDescription"
        )}
      </p>
      <ButtonLink
        className="display-inline-flex flex-align-center flex-justify-center flex-column margin-right-0 margin-top-3"
        href={appLogic.portalFlow.getNextPageRoute(
          "CONTINUE",
          {},
          { claim_id: claim.application_id }
        )}
      >
        <div>
          {t(
            "pages.claimsAdditionalUserNotFoundInfo.beginAdditionalNoEmployeeFoundInfoButton"
          )}
        </div>
      </ButtonLink>
    </React.Fragment>
  );
};

export default withBenefitsApplication(AdditionalUserNotFoundInfoBegin);
