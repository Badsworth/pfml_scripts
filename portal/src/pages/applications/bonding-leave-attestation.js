import Alert from "../../components/Alert";
import BackButton from "../../components/BackButton";
import ButtonLink from "../../components/ButtonLink";
import PropTypes from "prop-types";
import React from "react";
import Title from "../../components/Title";
import { Trans } from "react-i18next";
import { useTranslation } from "../../locales/i18n";
import withBenefitsApplication from "../../hoc/withBenefitsApplication";

export const BondingLeaveAttestation = (props) => {
  const { t } = useTranslation();
  const { appLogic, query } = props;
  return (
    <React.Fragment>
      <BackButton />
      <Title>{t("pages.claimsBondingLeaveAttestation.title")}</Title>
      <Trans
        i18nKey="pages.claimsBondingLeaveAttestation.lead"
        components={{ ul: <ul className="usa-list" />, li: <li /> }}
      />
      <Alert className="measure-6" state="info" noIcon>
        <p>{t("pages.claimsBondingLeaveAttestation.truthAttestation")}</p>
        <ButtonLink
          className="text-no-underline text-white"
          href={appLogic.portalFlow.getNextPageRoute("CONTINUE", {}, query)}
        >
          {t("pages.claimsBondingLeaveAttestation.submitApplicationButton")}
        </ButtonLink>
      </Alert>
    </React.Fragment>
  );
};

BondingLeaveAttestation.propTypes = {
  appLogic: PropTypes.shape({
    portalFlow: PropTypes.shape({
      getNextPageRoute: PropTypes.func.isRequired,
    }),
  }).isRequired,
  query: PropTypes.object.isRequired,
};

export default withBenefitsApplication(BondingLeaveAttestation);
