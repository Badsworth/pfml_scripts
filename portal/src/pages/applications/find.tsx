import withUser, { WithUserProps } from "../../hoc/withUser";
import InputText from "../../components/core/InputText";
import Lead from "../../components/core/Lead";
import PageNotFound from "../../components/PageNotFound";
import QuestionPage from "../../components/QuestionPage";
import React from "react";
import { isFeatureEnabled } from "../../services/featureFlags";
import useFormState from "../../hooks/useFormState";
import useFunctionalInputProps from "../../hooks/useFunctionalInputProps";
import { useTranslation } from "../../locales/i18n";

export const Find = (props: WithUserProps) => {
  const { t } = useTranslation();
  const { formState, updateFields } = useFormState();

  const getFunctionalInputProps = useFunctionalInputProps({
    appErrors: props.appLogic.appErrors,
    formState,
    updateFields,
  });

  const handleSubmit = async () => {
    await props.appLogic.benefitsApplications.associate(formState);
  };

  if (!isFeatureEnabled("channelSwitching")) return <PageNotFound />;

  return (
    <QuestionPage
      continueButtonLabel={t("pages.claimsAssociate.submitButton")}
      onSave={handleSubmit}
      title={t("pages.claimsAssociate.title")}
      titleSize="regular"
    >
      <Lead>{t("pages.claimsAssociate.lead")}</Lead>

      <InputText
        {...getFunctionalInputProps("tax_identifier_last4")}
        label={t("pages.claimsAssociate.taxIdLabel")}
        smallLabel
        width="small"
      />
      <InputText
        {...getFunctionalInputProps("absence_id")}
        label={t("pages.claimsAssociate.absenceIdLabel")}
        smallLabel
        width="medium"
      />
    </QuestionPage>
  );
};

export default withUser(Find);
