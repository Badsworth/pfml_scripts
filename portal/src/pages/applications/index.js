import Alert from "../../components/Alert";
import ApplicationCard from "../../components/ApplicationCard";
import ClaimCollection from "../../models/ClaimCollection";
import DashboardNavigation from "../../components/DashboardNavigation";
import Heading from "../../components/Heading";
import Lead from "../../components/Lead";
import PropTypes from "prop-types";
import React from "react";
import Title from "../../components/Title";
import { useTranslation } from "../../locales/i18n";
import withClaims from "../../hoc/withClaims";

/**
 * List of all applications associated with the authenticated user
 */
const Applications = (props) => {
  const { appLogic, claims, query } = props;
  const { t } = useTranslation();

  const hasClaims = claims.items.length > 0;
  const hasInProgressClaims = hasClaims && claims.inProgress.length > 0;
  const hasCompletedClaims = hasClaims && claims.completed.length > 0;

  return (
    <React.Fragment>
      {query && query.uploadedAbsenceId && (
        <Alert
          className="margin-bottom-3"
          heading={t("pages.applications.uploadSuccessHeading")}
          name="upload-success-message"
          state="success"
        >
          {t("pages.applications.uploadSuccessMessage", {
            absence_id: query.uploadedAbsenceId,
          })}
        </Alert>
      )}

      <DashboardNavigation activeHref={appLogic.portalFlow.pathname} />
      <Title hidden={hasClaims}>{t("pages.applications.title")}</Title>

      {!hasClaims && <p>{t("pages.applications.noClaims")}</p>}

      {hasInProgressClaims && (
        <React.Fragment>
          <div className="measure-6">
            <Lead>{t("pages.applications.claimsReflectPortal")}</Lead>
          </div>
          <Heading level="2" size="1">
            {t("pages.applications.inProgressHeading")}
          </Heading>
          {claims.inProgress.map((claim, index) => {
            return (
              <ApplicationCard
                appLogic={appLogic}
                key={claim.application_id}
                claim={claim}
                number={index + 1}
              />
            );
          })}
        </React.Fragment>
      )}

      {hasCompletedClaims && (
        <React.Fragment>
          <Heading level="2" size="1">
            {t("pages.applications.submittedHeading")}
          </Heading>
          {claims.completed.map((claim, index) => {
            return (
              <ApplicationCard
                appLogic={appLogic}
                key={claim.application_id}
                claim={claim}
                number={claims.inProgress.length + index + 1}
              />
            );
          })}
        </React.Fragment>
      )}
    </React.Fragment>
  );
};

Applications.propTypes = {
  appLogic: PropTypes.object.isRequired,
  claims: PropTypes.instanceOf(ClaimCollection).isRequired,
  query: PropTypes.shape({
    uploadedAbsenceId: PropTypes.string,
  }),
};

export default withClaims(Applications);
