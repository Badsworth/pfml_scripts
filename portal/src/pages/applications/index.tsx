import withBenefitsApplications, {
  WithBenefitsApplicationsProps,
} from "../../hoc/withBenefitsApplications";
import withUser, { WithUserProps } from "../../hoc/withUser";
import Alert from "../../components/core/Alert";
import ApplicationCard from "../../components/ApplicationCard";
import ButtonLink from "../../components/ButtonLink";
import Heading from "../../components/core/Heading";
import Lead from "../../components/core/Lead";
import PaginationNavigation from "src/components/PaginationNavigation";
import React, { useCallback, useMemo } from "react";
import Title from "../../components/core/Title";
import { Trans } from "react-i18next";
import isBlank from "src/utils/isBlank";
import routes from "../../routes";
import { useTranslation } from "../../locales/i18n";

interface PageQueryParam {
  name: string;
  value: number | null | string | string[];
}
interface IndexProps extends WithUserProps {
  query: { uploadedAbsenceId?: string };
}

/**
 * List of all applications associated with the authenticated user
 */
export const Index = (props: IndexProps) => {
  const { appLogic, query } = props;
  const { t } = useTranslation();
  const apiParams = {
    order_direction: "ascending",
    ...query,
  } as const;

  /**
   * Update the page's query string, to load a different page number,
   */
  const updatePageQuery = (paramsToUpdate: PageQueryParam[]) => {
    const params = new URLSearchParams(window.location.search);

    paramsToUpdate.forEach(({ name, value }) => {
      if (isBlank(value) || (typeof value !== "number" && value.length === 0)) {
        // Remove param if its value is null, undefined, empty string, or empty array
        params.delete(name);
      } else {
        params.set(name, value.toString());
      }
    });

    const paramsObj: { [key: string]: string } = {};
    for (const [paramKey, paramValue] of Array.from(params.entries())) {
      paramsObj[paramKey] = paramValue;
    }

    // Our withBenefitsApplication component watches the query string and
    // will trigger an API request when it changes.
    appLogic.portalFlow.updateQuery(paramsObj);
  };
  const PaginatedIndexPageWithBenefitsApplications = withBenefitsApplications(
    PaginatedIndexPage,
    apiParams
  );
  const componentSpecificProps = {
    updatePageQuery,
  };

  return (
    <React.Fragment>
      {query.uploadedAbsenceId && (
        <Alert
          className="margin-bottom-3"
          heading={t("pages.applications.uploadSuccessHeading")}
          state="success"
        >
          {t("pages.applications.uploadSuccessMessage", {
            absence_id: query.uploadedAbsenceId,
          })}
        </Alert>
      )}

      <div className="grid-row grid-gap-6">
        <PaginatedIndexPageWithBenefitsApplications
          appLogic={appLogic}
          {...componentSpecificProps}
        />
        <div className="desktop:grid-col-auto">
          <Heading level="2" className="usa-sr-only">
            {t("pages.applications.createApplicationHeading")}
          </Heading>

          <ButtonLink href={routes.applications.getReady} variation="outline">
            {t("pages.applications.getReadyLink")}
          </ButtonLink>
        </div>
      </div>
    </React.Fragment>
  );
};

interface PaginatedIndexPageProps extends WithBenefitsApplicationsProps {
  updatePageQuery: (params: PageQueryParam[]) => void;
}

const AppCard = (props) => {
  return (
    <ApplicationCard
      appLogic={props.appLogic}
      key={props.application_id}
      // @ts-expect-error PORTAL-1078
      claim={props.claim}
      number={props.number}
    />
  );
};

const PaginatedIndexPage = (props: PaginatedIndexPageProps) => {
  const { appLogic, claims, paginationMeta, updatePageQuery } = props;
  const { t } = useTranslation();

  // Redirect users who do not have claims
  // if (props.claims.isEmpty) {
  //   appLogic.portalFlow.goTo(routes.applications.getReady);
  //   return null;
  // }

  const hasClaims = claims.items.length > 0;
  const hasInProgressClaims = hasClaims && claims.inProgress.length > 0;
  const hasCompletedClaims = hasClaims && claims.completed.length > 0;

  /**
   * Event handler for when a next/prev pagination button is clicked
   */
  const handlePaginationNavigationClick = (pageOffset: number | string) => {
    updatePageQuery([
      {
        name: "page_offset",
        value: pageOffset,
      },
    ]);
  };

  return (
    <React.Fragment>
      <div className="desktop:grid-col">
        <Title>{t("pages.applications.title")}</Title>

        {hasInProgressClaims && (
          <React.Fragment>
            <div className="measure-6">
              <Lead>
                <Trans
                  i18nKey="pages.applications.claimsApprovalProcess"
                  components={{
                    "approval-process-link": (
                      <a
                        href={routes.external.massgov.approvalTimeline}
                        rel="noopener noreferrer"
                        target="_blank"
                      />
                    ),
                  }}
                />
              </Lead>
            </div>
            <Heading level="2">
              {t("pages.applications.inProgressHeading")}
            </Heading>
            {claims.inProgress.map((claim, index) => {
              return (
                <AppCard
                  key={claim.application_id}
                  appLogic={appLogic}
                  application_id={claim.application_id}
                  // @ts-expect-error PORTAL-1078
                  claim={claim}
                  number={index}
                />
              );
            })}
          </React.Fragment>
        )}

        {hasCompletedClaims && (
          <React.Fragment>
            <Heading level="2">
              {t("pages.applications.submittedHeading")}
            </Heading>
            {claims.completed.map((claim, index) => {
              return (
                <AppCard
                  key={claim.application_id}
                  appLogic={appLogic}
                  application_id={claim.application_id}
                  // @ts-expect-error PORTAL-1078
                  claim={claim}
                  number={index}
                />
              );
            })}
          </React.Fragment>
        )}
        {paginationMeta.total_pages > 0 && (
          <PaginationNavigation
            pageOffset={paginationMeta.page_offset}
            totalPages={paginationMeta.total_pages}
            onClick={handlePaginationNavigationClick}
          />
        )}
      </div>
    </React.Fragment>
  );
};

export default withUser(Index);
