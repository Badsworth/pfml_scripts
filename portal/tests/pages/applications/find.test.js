import { screen, waitFor } from "@testing-library/react";
import { Find } from "../../../src/pages/applications/find";
import { renderPage } from "../../test-utils";
import userEvent from "@testing-library/user-event";

describe("Find", () => {
  beforeEach(() => {
    process.env.featureFlags = {
      channelSwitching: true,
    };
  });

  it("renders page not found when feature flag isn't enabled", () => {
    process.env.featureFlags = { channelSwitching: false };
    renderPage(Find);

    expect(screen.getByText("Page not found")).toBeInTheDocument();
  });

  it("renders the page", () => {
    const { container } = renderPage(Find);
    expect(container).toMatchSnapshot();
  });

  it("associates application when user clicks submit", async () => {
    const associateMock = jest.fn();
    renderPage(Find, {
      addCustomSetup: (appLogic) => {
        appLogic.benefitsApplications.associate = associateMock;
      },
    });

    userEvent.type(
      screen.getByRole("textbox", { name: /social security/i }),
      "1234"
    );
    userEvent.type(
      screen.getByRole("textbox", { name: /application id/i }),
      "NTN-111-ABS-01"
    );
    userEvent.click(screen.getByRole("button", { name: "Continue" }));

    await waitFor(() => {
      expect(associateMock).toHaveBeenCalledWith({
        absence_id: "NTN-111-ABS-01",
        tax_identifier_last4: "1234",
      });
    });
  });
});
