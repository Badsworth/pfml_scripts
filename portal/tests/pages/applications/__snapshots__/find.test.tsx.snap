// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Find renders the page 1`] = `
<div>
  <form
    class="usa-form"
    method="post"
  >
    <h1
      class="js-title margin-top-0 margin-bottom-2 font-heading-lg line-height-sans-2"
      tabindex="-1"
    >
      Find My Application
    </h1>
    <p
      class="usa-intro"
    >
      To locate your application, please enter information for each field.
    </p>
    <div
      class="usa-form-group"
    >
      <label
        class="usa-label text-bold font-heading-xs measure-5"
        for="InputText1"
        id="InputText1_label"
      >
        Last 4 digits of Social Security Number
      </label>
      <input
        aria-labelledby="InputText1_label InputText1_hint InputText1_error"
        class="usa-input usa-input--small"
        id="InputText1"
        name="tax_identifier_last4"
        type="text"
        value=""
      />
    </div>
    <div
      class="usa-form-group"
    >
      <label
        class="usa-label text-bold font-heading-xs measure-5"
        for="InputText2"
        id="InputText2_label"
      >
        Application ID
      </label>
      <input
        aria-labelledby="InputText2_label InputText2_hint InputText2_error"
        class="usa-input usa-input--medium"
        id="InputText2"
        name="absence_id"
        type="text"
        value=""
      />
    </div>
    <button
      class="usa-button position-relative margin-top-4"
      type="submit"
    >
      Continue
    </button>
  </form>
</div>
`;
